11ty/search.njk

```diff
@@ -0,0 +1,83 @@
+---
+layout: layouts/search.njk
+---
+
+<header class="introduction">
+  <h1 class="introduction__title thicc-headline" id="title">
+    Search Results
+  </h1>
+</header>
+
+<section aria-live class="search">
+  <h2 id="search-title">Building your search results, one moment.</h2>
+  <ul id="search-results-list" class="u-no-padding-left list" aria-labelledby="search" />
+</section>
+
+{% block pageScript%}
+<script>
+  function getAllOccurrences(str, match) {
+    const matchLength = match.length;
+    const indices = []
+    let startIndex = 0, index
+
+    str = str.toLowerCase();
+    match = match.toLowerCase();
+
+    while ((index = str.indexOf(match, startIndex)) > -1) {
+        indices.push(index);
+        startIndex = index + matchLength;
+    }
+
+    return indices.map(index => {
+      // this needs some cleaning up
+      // currently there are edge cases that break the selection of the substring
+      const start = index < 50 ? index : index - 50;
+      const end = matchLength + 100;
+      const exerpt = str.substr(start, end)
+
+      return exerpt.replace(match, `<span class="search-results-list__list-item__highlight">${match}</span>`)
+    })
+  }
+
+  // don't use es6+ here because it will not be transpiled
+  window.addEventListener('DOMContentLoaded', function() {
+    function buildResultListItem(result, search) {
+      const occurrences = getAllOccurrences(result.content, search)
+      return `<h4><a href="${result.url}">${result.title}</a></h4>${occurrences.map(occurrence => `<p>...${occurrence}...</p>`).join('')}`
+    }
+
+    const init = async () => {
+      try {
+        const urlParams = new URLSearchParams(window.location.search);
+        const $searchTitle = document.querySelector('#search-title')
+        const $resultsContainer = document.querySelector('#search-results-list')
+
+
+        if (!urlParams.has('q')) {
+          $resultsContainer.innerHTML = ''
+          return
+        }
+
+        const value = urlParams.get('q')
+        const matches = await fetch(`https://climb-phi-staging.begin.app/self-defined-app-search?q=${value}`, {mode: 'cors'})
+        const results = await matches.json()
+
+        $resultsContainer.innerHTML = ''
+        $searchTitle.innerText = `We found ${results.length} ${results.length === 1 ? 'definition' : 'definitions'} containing your search term`
```

Comment:
```suggestion
        $searchTitle.innerText = `We found ${results.length} ${results.length === 1 ? 'definition' : 'definitions'} containing \u275d${value}\u275e`
```
- We should remind the user what they searched for 
- `\u275d` and `\u275e` are the unicode values for the chonky curly quotes. I picked them because I _think_ they appear elsewhere in the app. Example:
<img width="545" alt="Screenshot of SelfDefined app, reading, 'We found 5 definitions containing able" src="https://user-images.githubusercontent.com/13525251/89745984-05643d80-da6c-11ea-80e1-1d89feb9b85b.png">